File: features/favorites/src/main/java/com/vp/favorites/viewmodel/FavouriteListViewModel.java
package com.vp.favorites.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import co.uk.missionlabs.db.FavouritesDB;
import co.uk.missionlabs.db.Model.Favourite;
import co.uk.missionlabs.db.Model.ListItem;


public class FavouriteListViewModel extends ViewModel {
    FavouritesDB db = new FavouritesDB();
    private MutableLiveData<List<ListItem>> liveData = new MutableLiveData<>();

    @Inject
    FavouriteListViewModel() {

    }

    public LiveData<List<ListItem>> observeFavourites() {
        return liveData;
    }


    public void getFavourites() {
        ArrayList<ListItem> listItems = new ArrayList<>();
        ArrayList<Favourite> favourites = db.getFavourites();
        for(int i=0;i<favourites.size();i++){
            ListItem item = new ListItem();
            item.setItem(favourites.get(i).getMovieDetail().getTitle(),favourites.get(i).getMovieDetail().getYear(),favourites.get(i).getImdbID(),favourites.get(i).getMovieDetail().getPoster());
            listItems.add(item);
        }

        liveData.postValue(listItems);

    }
}

File: features/favorites/src/main/java/com/vp/favorites/viewmodel/FavouriteListViewModel.kt
package com.vp.favorites.viewmodel

import androidx.lifecycle.*
import com.vp.favorites.model.FavouriteListDataRepository
import com.vp.favorites.model.FavouriteListState
import com.vp.list.model.ListItem
import com.vp.list.viewmodel.ListState
import io.reactivex.disposables.Disposable
import javax.inject.Inject

class FavouriteListViewModel @Inject internal constructor(
        private val dataRepository: FavouriteListDataRepository
) : ViewModel(), LifecycleObserver {

    val favouriteListItems: LiveData<List<ListItem>> = Transformations.map(dataRepository.liveData) { it.items }

    val favouriteListState: LiveData<FavouriteListState> = Transformations.map(dataRepository.liveData) { it.listState }

    private var disposable: Disposable? = null
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun create() {
        disposable =dataRepository.fetchData()
    }
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun destroy() {
       disposable?.run { if (!isDisposed) dispose() }
    }
}